src/Components/CutPlaneMenu.jsx

```diff
@@ -114,10 +121,147 @@ export default function CutPlaneMenu() {
+        ifcModel?.geometry.boundingBox.min.x) /
+      2,
+      (ifcModel?.geometry.boundingBox.max.y +
+        ifcModel?.geometry.boundingBox.min.y) /
+      2,
+      (ifcModel?.geometry.boundingBox.max.z +
+        ifcModel?.geometry.boundingBox.min.z) /
+      2,
+  )
+}
+
+
+/**
+ * getElementBoundingBox creates a bounding box around the model
+ *
+ * @param {object} selection seclected meshes
+ * @return {object} boudingBox geometry
+ */
+export function getElementBoundingBox(selection) {
+  const geometry = new BufferGeometry()
+  const coordinates = []
+  const alreadySaved = new Set()
+  const position = selection.geometry.attributes['position']
+  const vertices = Float32Array.from(coordinates)
+  const mesh = new Mesh(geometry)
+  const boundingBox = new Box3()
+  geometry.setAttribute('position', new BufferAttribute(vertices, selection.geometry.index.count))
+  boundingBox.setFromObject(mesh)
+
+  for (let i = 0; i < selection.geometry.index.array.length; i++) {
+    if (!alreadySaved.has(selection.geometry.index.array[i])) {
+      coordinates.push(position.getX(selection.geometry.index.array[i]))
+      coordinates.push(position.getY(selection.geometry.index.array[i]))
+      coordinates.push(position.getZ(selection.geometry.index.array[i]))
+      alreadySaved.add(selection.geometry.index.array[i])
+    }
+  }
+
+  return boundingBox
+}
+
+/**
+ * toggleClippingPlane turns clipping plane on and off
+ *
+ * @param {object} viewer bouding box
+ */
+export function removePlanes(viewer) {
+  viewer?.clipper.deleteAllPlanes()
+  const clippingPlanes = viewer?.clipper['context'].clippingPlanes
+  for (const plane of clippingPlanes) {
+    viewer?.clipper['context'].removeClippingPlane(plane)
+  }
+}
+
+
+/**
+ * helper method to get the location of all of the planes currently in the scene
+ *
+ * @param {object} viewer
+ * @param {object} model
+ * @return {object} offsetObj contins plane normal key and
+ */
+export function getPlaneOffset(viewer, ifcModel) {
+  if (viewer.clipper.planes.length > 0) {
+    let planeNormal
+    let planeAxisCenter
+    let planeOffsetFromCenter
+    let planeHash
+    const planeOffsetFromModelBoundary = viewer.clipper.planes[0].plane.constant
+    const modelCenter = getModelCenter(ifcModel)
+    for (const [key, value] of Object.entries(viewer.clipper.planes[0].plane.normal)) {
+      if (value !== 0 ) {
+        planeNormal = key
+        planeAxisCenter = modelCenter[planeNormal]
+        planeOffsetFromCenter = planeOffsetFromModelBoundary - planeAxisCenter
+        const planeOffsetFromCenterTrim = planeOffsetFromCenter
+        planeHash = `${planeNormal},${planeOffsetFromCenterTrim}`
+      }
+    }
+    const planeOffsetObj = {planeAxis: planeHash}
+    return planeOffsetObj
+  }
+  return null
+}
+
+
+/**
+ * helper method to get the location of all of the planes currently in the scene
+ *
+ * @param {object} viewer
+ * @param {object} model
+ * @return {object} add plane to url
```

Comment:
Should there be a return value on this method?  I don't see a return value for addHashParams...